### DISCUSSION TITLE
12 Hour vs 24 hour Clock

### DISCUSSION THREAD

**Discussion by rcblackwell** [BigBobbas/ESP32-S3-Box3-Custom-ESPHome#68] [Ideas]
Rob,

Members of my family have a hard time understanding the 24 hour clock format. I'm willing to undertake the task of adding code that would provide a selectable 12/24 hour clock format. Question is; where should the selection menu be placed? I think it should be an item in the Misc Settings menu. What are your thoughts?

**Comment by BigBobbas** [comment]
Hi Robert,
you are a true star! that would be awesome and yes the misc settings would be the best place I think. 
Thank you so much for your help with everything 👍

**Reply by rcblackwell** [reply]
My pleasure! It's a great project that's helping me learn YAML as well.

I'll get started later today. I doubt my work will be finished prior to leaving for vacation Aug 7th. If not, I'll dive right back in when I return August 24th. Until then, keep up the amazing work.

**Comment by rcblackwell** [comment]
I've been slower getting at this than I anticipated. I had trouble compiling the code on my HA Blue and had to learn how to build an ESPHome server on Windows WSL Ubuntu.. Building the server wasn't a big issue. There's lots of online help for that. What I had trouble with was installing compiled code on devices. For the life of me, I couldn't get mDNS to work on the server. After several head banging exercises over the course of a week, I finally determined that the virtual network adapter had to be configured differently. Once that was sorted, I was back in business.

I'm ready to move ahead on this, if you're okay with me doing so?

**Reply by BigBobbas** [reply]
Glad you got it sorted and hope that you notice the benefit. 
I give up with mDNS and usually just add the ip to the config as it's so hit and miss.
Yes that would be great if you would like to add it. I have created a 'preferences' page under settings and am working on it right now.  I think it would probably be best suited there, but open to suggestions.
It is still in progress, however I would like to think I will have it uploaded by tomorrow evening (uk time)  as I need to get some sleep soon :)

**Reply by rcblackwell** [reply]
The Box 3 code takes nearly thirty minutes on an HA Blue to compile. On wsl Ubuntu ESPHome, it takes about five. I'd say that's a significant improvement!

It looks like there's been some great addition's since I last reviewed the DEV code! The preference page does look like the right place to add the 12/24hr setting. I have parts of the required code written. I'll be working on the balance this week. I'm hopeful I'll have something to post by late Friday, EST.

**Reply by rcblackwell** [reply]
Rob

My code is ready to insert. Which one of the files should I add it to. Would the file at [s3b_dev.yaml](https://github.com/BigBobbas/ESP32-S3-Box3-Custom-ESPHome/blob/dev-live-wip/s3b_dev.yaml) be the correct one?

**Reply by BigBobbas** [reply]
Hi Robert
yes that would be great. Thank you
